### PR TITLE
SWS-327 Fix fetch Istio Rule with non supported handlers/instances

### PR DESCRIPTION
**Pull request by lucasponce**
- A non supported handler/instance is converted in a nil object but prevent to fail

### CODE REVIEW

**Comment by lucasponce** [comment]
@xeviknal could you review ?
This is to fix a missing validation.
Thx.

File: kubernetes/istio_rules_service.go [hunk 1]
```diff
@@ -103,7 +104,13 @@ func (in *IstioClient) getActionDetails(namespace string, istiorule string, acti
...
 
 func (in *IstioClient) getActionHandler(namespace string, handlerType string, handlerName string, handlerChan chan<- istioResponse) {
-	result, err := in.istio.Get().Namespace(namespace).Resource(istioTypePlurals[handlerType]).SubResource(handlerName).Do().Get()
+	handlerTypePlural, ok := istioTypePlurals[handlerType]
+	if !ok {
+		log.Warningf("Handler type %s is not supported", handlerType)
+		handlerChan <- istioResponse{}
+		return
+	}
+	result, err := in.istio.Get().Namespace(namespace).Resource(handlerTypePlural).SubResource(handlerName).Do().Get()
```
Comment (jotak) [inline kubernetes/istio_rules_service.go]: should not the error be checked and reported here? (I see you just moved these lines so the problem (if any) was already there)

File: kubernetes/istio_rules_service.go [hunk 2]
```diff
@@ -127,7 +134,13 @@ func (in *IstioClient) getActionInstance(namespace string, istiorule string, ins
...
 		return
 	}
-	result, err := in.istio.Get().Namespace(namespace).Resource(istioTypePlurals[instanceType]).SubResource(instanceName).Do().Get()
+	istioTypePlural, ok := istioTypePlurals[instanceType]
+	if !ok {
+		log.Warningf("Instance type %s is not supported", instanceType)
+		instancesChan <- istioResponse{}
+		return
+	}
+	result, err := in.istio.Get().Namespace(namespace).Resource(istioTypePlural).SubResource(instanceName).Do().Get()
```
Comment (jotak) [inline kubernetes/istio_rules_service.go]: same here

File: kubernetes/istio_rules_service.go (hunk 1 above)
Comment (lucasponce) [reply kubernetes/istio_rules_service.go]: No, this is a case where there is not yet support for this type.
So, the preferred behaviour is to return an empty type.

File: kubernetes/istio_rules_service.go (hunk 2 above)
Comment (lucasponce) [reply kubernetes/istio_rules_service.go]: Answered in the previous comment.

**Pull request merged**